lib/utils/matrix_sdk_extensions/matrix_file_extension.dart

```diff
@@ -45,22 +42,19 @@ extension MatrixFileExtension on MatrixFile {
     return;
   }
 
-  Future<String> downloadFileInWeb(BuildContext context) async {
+  Future<String> downloadFileInWeb() async {
     Logs().d("MatrixFileExtension()::downloadFileInWeb()::download on Web");
 
-    final directory = await FileSaver.instance.saveFile(
-      name,
-      bytes!,
-      extensionFromMime(mimeType),
-      mimeType: mimeType.toMimeTypeEnum(),
-    );
+    final FileSaveLocation? result = await getSaveLocation(suggestedName: name);
+    if (result == null) {
+      // Operation was canceled by the user.
+      return "";
+    }
 
-    TwakeSnackBar.show(
-      context,
-      L10n.of(context)!.downloadFileInWeb(directory),
-    );
+    final XFile file = XFile.fromData(bytes!, mimeType: mimeType, name: name);
+    await file.saveTo(result.path);
 
-    return '$directory/$name';
+    return '';
```

Comment:
so we don need the result `String`?

Reply:
It is not currently used and in web the browser can't know where a file is saved for security reasons

Reply:
https://github.com/linagora/twake-on-matrix/issues/1387#issuecomment-1970875195

Reply:
so if this function return a empty string, why not changes it to `void`

Reply:
done